source/common/vb_gui.c

```diff
@@ -343,14 +347,38 @@ int emulation_reset(void) {
     return D_EXIT;
 }
 
-int emulation_sstate(void) {
+char * get_savestate_path(int state) {
+    char * sspath;
+    sspath = (char *) malloc(131 * sizeof(char));
+    char file_name[8];
+    char * last_slash = strrchr(tVBOpt.ROM_PATH, '/');
+    strncpy(sspath, tVBOpt.ROM_PATH, last_slash - tVBOpt.ROM_PATH);
+    sspath[last_slash - tVBOpt.ROM_PATH] = '\0';
+    strcat(sspath, "/vb_savestates");
+    if(stat(sspath, &st) == -1)
+        mkdir(sspath, 0777);
+    strcat(sspath, tVBOpt.ROM_PATH + (last_slash - tVBOpt.ROM_PATH));
+    sspath[strlen(sspath) - 3] = '\0';
+    if(stat(sspath, &st) == -1)
+        mkdir(sspath, 0777);
+    sprintf(file_name, "/%d.rds", state);
+    strcat(sspath, file_name);
+
+    return sspath;
+}
+
+int emulation_rmstate(int state) {
+    char* sspath = get_savestate_path(state);
+
+    return remove(sspath);
+}
+
+int emulation_sstate(int state) {
     int i;
     int highbyte;
     int lowbyte;
-    char sspath[131];
     FILE* state_file;
-
-    sprintf(sspath, "%s.rds", tVBOpt.ROM_NAME);
+    char* sspath = get_savestate_path(state);
```

Comment:
When will the memory be released?

Reply:
addressed in [1561bf](https://github.com/Piiman/red-viper/commit/18561bffe44680dbee11116bd3e8412ae1ec0f1a)